chaosLib/litmus/aws_az_chaos/lib/aws_az_chaos.py

```diff
@@ -0,0 +1,150 @@
+		targetZones = experimentsDetails.LoadBalancerZones.split(",")
+		
+		logging.info("[Info]: Target available zone list, %s", targetZones)
+
+		if experimentsDetails.EngineName != "" :
+			msg = "Injecting " + experimentsDetails.ExperimentName + " chaos on available zone"
+			types.SetEngineEventAttributes(eventsDetails, types.ChaosInject, msg, "Normal", chaosDetails)
+			events.GenerateEvents(eventsDetails, chaosDetails, "ChaosEngine", clients)
+		
+		# Detaching the target zones from loa balancer 
+		for azone in targetZones:
+
+			logging.info("[Info]: Detaching the following zone(s), Zone Name : %s", azone)
+			err = statusAws.detachAZfromLB(experimentsDetails, azone)
+			if err != None:
+				return err
+			if chaosDetails.Randomness:
+				err = common.RandomInterval(experimentsDetails.ChaosInterval)
+				if err != None:
+					return err
+			else:
+				#Waiting for the chaos interval after chaos injection
+				if experimentsDetails.ChaosInterval != "":
+					logging.info("[Wait]: Wait for the chaos interval %s",(experimentsDetails.ChaosInterval))
+					waitTime = maths.atoi(experimentsDetails.ChaosInterval)
+					common.WaitForDuration(waitTime)
+
+			# Attaching the target available zone after the chaos injection
+			logging.info("[Status]: Atach the available zone back to load balancer")
+			err = statusAws.attachAZtoLB(experimentsDetails, azone)
+			if err != None:
+				return err
+			
+   			#Verify the status of available zone after the chaos injection
+			logging.info("[Status]: Checking AWS load balancer's AZ status")		
+			err = statusAws.CheckAWSStatus(experimentsDetails)
+			if err != None:
+				return err
+			
+			duration = (datetime.now() - ChaosStartTimeStamp).seconds
+
+	logging.info("[Completion]: %s chaos is done",(experimentsDetails.ExperimentName))
+
+	return None
+
+# injectChaosInParallelMode disable the target available zone from loadbalancer in parallel mode (all at once)
+def injectChaosInParallelMode(experimentsDetails , chaosDetails , eventsDetails , resultDetails, clients, statusAws):
+	
+
+	#ChaosStartTimeStamp contains the start timestamp, when the chaos injection begin
+	ChaosStartTimeStamp = datetime.now()
+	duration = (datetime.now() - ChaosStartTimeStamp).seconds
+	
+	while duration < experimentsDetails.ChaosDuration:
+		
+  		# Get the target available zone details for the chaos execution
+		targetZones = experimentsDetails.LoadBalancerZones.split(",")
+		logging.info("[Info]: Target available zone list, %s", targetZones)
+			
+		if experimentsDetails.EngineName != "" :
+			msg = "Injecting " + experimentsDetails.ExperimentName + " chaos on available zone"
+			types.SetEngineEventAttributes(eventsDetails, types.ChaosInject, msg, "Normal", chaosDetails)
+			events.GenerateEvents(eventsDetails, chaosDetails, "ChaosEngine",clients)
+		
+		# Detaching the target zones from loa balancer
+		for azone in targetZones:
+			logging.info("[Info]: Detaching the following zone(s), Zone Name %s", azone)
```

Comment:
```suggestion
			logging.info("[Info]: Detaching the following zone, Zone Name %s", azone)
```

Reply:
same

Reply:
Since we are looping through a list of comma-separated zones, I guess in every iteration `azone` will consist of only one zone name.

Reply:
Let me move this info, outside the for loop